ext/net/ops.rs

```diff
@@ -674,11 +675,21 @@ where
 
   lookup
     .map_err(|e| match e.kind() {
-      ResolveErrorKind::NoRecordsFound { .. } => NetError::DnsNotFound(e),
-      ResolveErrorKind::Message("No connections available") => {
+      ResolveErrorKind::Proto(ProtoError { kind, .. })
+        if matches!(**kind, ProtoErrorKind::NoRecordsFound { .. }) =>
+      {
+        NetError::DnsNotFound(e)
+      }
+      ResolveErrorKind::Proto(ProtoError { kind, .. })
```

Comment:
I agree that this is the correct change -- in fact, the previous version of the match arm (for `ResolveErrorKind::Message("No connections available")`, originally from 25771b3d9b36a7262d42809bfbb5497ed8057780) hasn't been effective for a while now (a `NoConnections` variant, first in `ResolveErrorKind`, now in `ProtoErrorKind`, has been around since 0.21, or Feb 2022).